### PR TITLE
PD-4368 Advanced examples

### PR DESCRIPTION
**Pull request by agaffney**

### CODE REVIEW

File: docs/examples/site_config.yml [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+          default: 512
+        disk:
+          default: 10
+        instances:
+          default: 3
+        # Default constraints make sure service instances are spread out across availability zones
+        constraints:
+          default:
+            - [ "@region", "IS", "aws/us-east-1" ]
+            - [ "@zone", "GROUP_BY", "3" ]
```
Comment (wolf31o2) [inline docs/examples/site_config.yml]: Comment these, but leave them, mentioning that fault-domain detection is an Enterprise feature

File: docs/examples/site_config.yml [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                - name: VIP_{{ port_index }}
+                  value: /{{ APP.id }}:{{ port.vip_port }}
+                  condition: port.vip_port
+        # Automatically add labels for service group/name
+        container_labels:
+          default_action: prepend
+          default:
+            - service={{ APP.group }}
+            - service_component={{ APP.name }}
+        # Provide fetch config for ECR auth
```
Comment (wolf31o2) [inline docs/examples/site_config.yml]: append `with ecr-login package`

File: docs/examples/site_config.yml [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            minimum_health_capacity:
+              default: 1
+            maximum_over_capacity:
+              default: 1
+        secrets:
+          fields:
+            source:
+              default: '{{ secret.name }}'
+              transform:
+                prefix: '{{ VARS.SERVICE_NAME }}/'
```
Comment (wolf31o2) [inline docs/examples/site_config.yml]: `secrets` here?

File: docs/examples/site_config.yml [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+                    config:
+                      remove:
+                        headers:
+                          - "X-Target-App"
+                      append:
+                        headers:
+                          # There's no space after the colon to match the previous behavior
+                          - "X-Target-App:{{ proxy.target_app }}"
+                # Include keycloak plugin when 'keycloak' param is True
+                - name: "keycloak"
```
Comment (wolf31o2) [inline docs/examples/site_config.yml]: The `keycloak` plugin isn't OSS (yet?).

Comment (agaffney) [reply docs/examples/site_config.yml]: Removed

File: docs/examples/site_config.yml (hunk 3 above)
Comment (agaffney) [reply docs/examples/site_config.yml]: This is the generic Marathon/Metronome `secrets` stuff for DC/OS enterprise, so this is fine to stay.

**Pull request merged**